src/classes/RD2_ValidationService_TEST.cls

```diff
@@ -200,6 +212,208 @@ private with sharing class RD2_ValidationService_TEST {
         System.assert(String.isBlank(errMessage), 'Recurring Donation should be created successfully: ' + errMessage);
     }
 
+
+    /***
+    * @description Verifies an exception is thrown when Amount is not specified, zero or a negative number on an Active Recurring Donation.
+    */
+    @isTest
+    private static void shouldNotCreateActiveRDWhenAmountIsNotValid() {
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        Contact con = getContact();
+
+        TEST_RecurringDonationBuilder rdBuilder = new TEST_RecurringDonationBuilder()
+            .withContact(con.Id)
+            .withDefaultValues()
+            .withDayOfMonth('15');
+
+        for (Decimal amount : new Set<Decimal>{
+            null,
+            0,
+            -10
+        }) {
+            npe03__Recurring_Donation__c rd = rdBuilder
+                .withAmount(amount)
+                .build();
+
+            String errMessage = '';
+            try {
+                insert rd;
+            } catch (Exception e) {
+                errMessage = e.getMessage();
+            }
+            System.assert(errMessage.contains(System.Label.RD2_AmountMustBeValid),
+                'Amount validation error should be expected: ' + errMessage);
+        }
+    }
+
+    /***
+    * @description Verifies if it creates Recurring donations for closed status when
+        invalid amount is specified.
+    */
+    @isTest
+    private static void shouldCreateRDForClosedRDWhenAmountIsNotValid() {
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        Contact con = getContact();
+
+        TEST_RecurringDonationBuilder rdBuilder = new TEST_RecurringDonationBuilder()
+            .withContact(con.Id)
+            .withDefaultValues()
+            .withDayOfMonth('15')
+            .withStatusClosed();
+
+        for (Decimal amount : new Set<Decimal>{
+            null,
+            0,
+            -10
+        }) {
+            npe03__Recurring_Donation__c rd = rdBuilder
+                .withAmount(amount)
+                .build();
+
+            String errMessage = '';
+            try {
+                insert rd;
+            } catch (Exception e) {
+                errMessage = e.getMessage();
+            }
+            System.assert(String.isBlank(errMessage), 'Recurring Donation should be created successfully: ' + errMessage);
+        }
+    }
+
+    /***
+    * @description Verifies if it displays error message when installment period
+        is monthly and day of month is blank
+    */
+    @isTest
+    private static void shouldFailWhenDayOfMonthIsBlankForMonthlyInstallmentPeriod() {
+        String errMessage = '';
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        Contact con = UTIL_UnitTestData_TEST.getContact();
```

Comment:
Use `getContact()` here or delete contact variable and create rd `withContact(getContact().Id)`. This and other similar tests in this class should be changed.